patches/api/0423-Add-drops-to-shear-events.patch

```diff
@@ -0,0 +1,111 @@
++    }
++    @org.jetbrains.annotations.ApiStatus.Internal
++    public BlockShearEntityEvent(@NotNull Block dispenser, @NotNull Entity sheared, @NotNull ItemStack tool, @NotNull java.util.List<ItemStack> drops) {
++        // Paper end
+         super(dispenser);
+         this.sheared = sheared;
+         this.tool = tool;
++        this.drops = drops; // Paper
+     }
+ 
+     /**
+@@ -64,4 +73,26 @@ public class BlockShearEntityEvent extends BlockEvent implements Cancellable {
+     public static HandlerList getHandlerList() {
+         return handlers;
+     }
++    // Paper start - drops API
++    /**
++     * Get a list of drops for this shearing.
++     * <p>
++     * The mutability of this list is undefined so
++     * use {@link #setDrops(java.util.List)} to change the drops.
++     *
++     * @return the shearing drops
++     */
++    public java.util.@NotNull List<ItemStack> getDrops() {
++        return this.drops;
++    }
++
++    /**
++     * Sets the drops for the shearing.
++     *
++     * @param drops the shear drops
++     */
++    public void setDrops(java.util.@NotNull List<org.bukkit.inventory.ItemStack> drops) {
++        this.drops = drops;
++    }
++    // Paper end - drops API
+ }
+diff --git a/src/main/java/org/bukkit/event/player/PlayerShearEntityEvent.java b/src/main/java/org/bukkit/event/player/PlayerShearEntityEvent.java
+index 04b3dce008edefb045162d0f69f87462ea1f3534..debd7bfaed87d7cc93422f7f5bb008c0d52568ad 100644
+--- a/src/main/java/org/bukkit/event/player/PlayerShearEntityEvent.java
++++ b/src/main/java/org/bukkit/event/player/PlayerShearEntityEvent.java
+@@ -18,12 +18,21 @@ public class PlayerShearEntityEvent extends PlayerEvent implements Cancellable {
+     private final Entity what;
+     private final ItemStack item;
+     private final EquipmentSlot hand;
++    private java.util.List<ItemStack> drops; // Paper
+ 
++    @Deprecated // Paper - include drops
+     public PlayerShearEntityEvent(@NotNull Player who, @NotNull Entity what, @NotNull ItemStack item, @NotNull EquipmentSlot hand) {
++        // Paper start
++        this(who, what, item, hand, java.util.Collections.emptyList());
++    }
++    @org.jetbrains.annotations.ApiStatus.Internal
++    public PlayerShearEntityEvent(@NotNull Player who, @NotNull Entity what, @NotNull ItemStack item, @NotNull EquipmentSlot hand, @NotNull java.util.List<ItemStack> drops) {
++        // Paper end
+         super(who);
+         this.what = what;
+         this.item = item;
+         this.hand = hand;
++        this.drops = drops; // Paper
+     }
+ 
+     @Deprecated
+@@ -82,4 +91,26 @@ public class PlayerShearEntityEvent extends PlayerEvent implements Cancellable {
+         return handlers;
+     }
+ 
++    // Paper start - drops API
++    /**
++     * Get the list of drops for this shearing.
++     * <p>
++     * The mutability of this list is undefined so
++     * use {@link #setDrops(java.util.List)} to change the drops.
```

Comment:
This list could be unmodifiable to be sure to have at least a consistent behavior in all cases